src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
+            ++tail;
+          }
 #elif defined(MSWIN)
```

Comment:
> Far as I can tell, the windows specific control was never introduced into the neovim codebase.

@pbhandari is right, I had to go back into the vim repo (see my link above) to find the window enabling code path. With windows support on its way, we need this, but also need to check what else was automaticlly excluded.